b2sdk/sync/sync.py

```diff
@@ -145,13 +145,13 @@ def _validate(self):
         if self.compare_threshold < 0:
             raise InvalidArgument('compare_threshold', 'must be a positive integer')
 
-        if not isinstance(self.newer_file_mode, NewerFileSyncMode):
+        if self.newer_file_mode not in list(NewerFileSyncMode):
```

Comment:
using `list` here suggests that the structure is mutable. Generally use `tuple` by default (actually `frozenset`, not `tuple`, because the order is not important), but here `NewerFileSyncMode.__members__` would be cleaner. I think it is ugly, but that's how standard library says we should deal with enums, so lets not invent a new way.

Reply:
https://docs.python.org/3/library/enum.html#iteration

```
In [1]: from enum import Enum                                                                                                                 

In [2]: class A(Enum): 
   ...:     RED=1 
   ...:     BLUE=2 
   ...:                                                                                                                                       

In [3]: A.RED                                                                                                                                 
Out[3]: <A.RED: 1>

In [4]: A.RED in A                                                                                                                            
Out[4]: True

In [5]: A.RED in A.__members__                                                                                                                
Out[5]: False

In [6]: A.RED in list(A)                                                                                                                      
Out[6]: True

In [7]: A.__members__                                                                                                                         
Out[7]: mappingproxy({'RED': <A.RED: 1>, 'BLUE': <A.BLUE: 2>})

In [8]: 'RED' in A.__members__                                                                                                                
Out[8]: True
```

Reply:
Uhh. Tuple then!